packages/debug/service.go

```diff
@@ -0,0 +1,154 @@
+package debug
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/k8s-ces-control/packages/doguinteraction"
+	"github.com/sirupsen/logrus"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	pbMaintenance "github.com/cloudogu/k8s-ces-control/generated/maintenance"
+	"github.com/cloudogu/k8s-ces-control/generated/types"
+)
+
+const (
+	maintenanceTitle          = "Service unavailable"
+	activateMaintenanceText   = "Activating debug mode"
+	deactivateMaintenanceText = "Deactivating debug mode"
+	logLevelDebug             = "DEBUG"
+	interErrMsg               = "internal error"
+)
+
+type defaultDebugModeService struct {
+	pbMaintenance.UnimplementedDebugModeServer
+	globalConfig          configurationContext
+	doguConfig            doguRegistry
+	clientSet             clusterClientSet
+	debugModeRegistry     debugModeRegistry
+	maintenanceModeSwitch maintenanceModeSwitch
+	namespace             string
+	doguInterActor        doguInterActor
+}
+
+// NewDebugModeService returns an instance of debugModeService.
+func NewDebugModeService(registry cesRegistry, clusterClient clusterClientSet, namespace string) *defaultDebugModeService {
+	cmDebugModeRegistry := NewConfigMapDebugModeRegistry(registry, clusterClient, namespace)
+	globalConfig := registry.GlobalConfig()
+	return &defaultDebugModeService{
+		globalConfig:          globalConfig,
+		doguConfig:            registry.DoguRegistry(),
+		clientSet:             clusterClient,
+		debugModeRegistry:     cmDebugModeRegistry,
+		maintenanceModeSwitch: NewDefaultMaintenanceModeSwitch(globalConfig),
+		namespace:             namespace,
+		doguInterActor:        doguinteraction.NewDefaultDoguInterActor(clusterClient, namespace, registry),
+	}
+}
+
+// Enable enables the debug mode, sets dogu log level to debug and restarts all dogus.
+func (s *defaultDebugModeService) Enable(ctx context.Context, req *pbMaintenance.ToggleDebugModeRequest) (*types.BasicResponse, error) {
+	err := s.maintenanceModeSwitch.ActivateMaintenanceMode(maintenanceTitle, activateMaintenanceText)
+	if err != nil {
+		return nil, createInternalError(fmt.Errorf("failed to activate maintenance mode: %w", err))
+	}
+
+	defer func() {
+		err = s.maintenanceModeSwitch.DeactivateMaintenanceMode()
+		if err != nil {
+			logrus.Error(fmt.Errorf("failed to deactivate maintenance mode: %w", err), interErrMsg)
+		}
+	}()
+
+	err = s.debugModeRegistry.Enable(ctx, req.Timer)
+	if err != nil {
+		return nil, createInternalError(fmt.Errorf("failed to enable debug mode registry: %w", err))
+	}
+
+	err = s.debugModeRegistry.BackupDoguLogLevels(ctx)
+	if err != nil {
+		return nil, createInternalError(fmt.Errorf("failed to backup dogu log levels: %w", err))
+	}
+
+	err = s.doguInterActor.SetLogLevelInAllDogus(logLevelDebug)
```

Comment:
should we do a rollback if any of these steps fail?
For example if one on the dogus could not be stopped, the method returns an error, but all previously stopped dogus will stay stopped.